beacon_node/eth2-libp2p/src/rpc/handler.rs

```diff
@@ -277,18 +290,18 @@ where
 
     /// Updates the `KeepAlive` returned by `connection_keep_alive`.
     ///
-    /// The handler stays alive as long as there are inbound/outbound substreams established and no
-    /// items dialing/to be dialed. Otherwise it is given a grace period of inactivity of
+    /// Checks if the handler is idle, if so it is given a grace period of inactivity of
     /// `self.inactive_timeout`.
     fn update_keep_alive(&mut self) {
-        // Check that we don't have outbound items pending for dialing, nor dialing, nor
-        // established. Also check that there are no established inbound substreams.
-        let should_shutdown = self.dial_queue.is_empty()
-            && self.dial_negotiated == 0
-            && self.outbound_substreams.is_empty()
-            && self.inbound_substreams.is_empty();
-
-        if should_shutdown {
+        if self.idle() {
+            if let KeepAlive::Until(when) = self.keep_alive {
+                // we were already on a idle state, if we are inside the inactivity period do not
+                // prolong it
+                if when > Instant::now() {
```

Comment:
The Swarm should check the Handler's keep alive. If `Until` is expired, it will close the connection. We don't have to update it within the handler. 
We only have to update the Until if we want to extend the time